### PR TITLE
[SYCL] Add some trivial util functions for half type.

### PR DESCRIPTION
**Pull request by jinge90**
The usage of sycl::half and CUDA half is different, sycl::half overrides arithmetic and comparison operators(+, -, *, /, >, <,==.!=...) while CUDA math provides those arithmetic and comparison functionalities via a bunch of util functions. This PR provides some sycl::half util functions aligning with CUDA math, users who are porting half related CUDA code to SYCL will spend less effort with those util functions' help.
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/intel/math/imf_half_trivial.hpp [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+sycl::half hneg(sycl::half x) { return -x; }
+
+sycl::half hsub(sycl::half x, sycl::half y) { return x - y; }
+
+sycl::half hsub_sat(sycl::half x, sycl::half y) {
+  return sycl::clamp((x - y), sycl::half(0.f), sycl::half(1.0f));
+}
+
+sycl::half hdiv(sycl::half x, sycl::half y) { return x / y; }
```
Comment (akolesov-intel) [inline sycl/include/sycl/ext/intel/math/imf_half_trivial.hpp:47]: Could you please also add another trivial function: reciprocal 'hrcp':

sycl::half hrcp(sycl::half x) { return 1.0f / x; }

**Review by akolesov-intel — CHANGES_REQUESTED** [review]

File: sycl/include/sycl/ext/intel/math/imf_half_trivial.hpp (hunk 1 above)
Comment (jinge90) [reply sycl/include/sycl/ext/intel/math/imf_half_trivial.hpp:47]: We have already had it and will map hrcp to corresponding _imf_* functions.

**Comment by jinge90** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1250

**Review by akolesov-intel — APPROVED** [review]

**Comment by jinge90** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1250

**Comment by jinge90** [comment]
Hi, @xtian-github , @zettai-reido and @intel/llvm-reviewers-runtime 
Could you take a look at this PR?
Thanks very much.

**Review by xtian-github — APPROVED** [review]
LGTM

**Review by zettai-reido — APPROVED** [review]
LGTM

**Comment by romanovvlad** [comment]
Hi,

Since the functions are in public namespace could you please clarify if we have an extension document covering them? If not, shouldn't we add/update existing one?
Could you please add more details to the PR summary?

**Comment by jinge90** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1250

**Review by xtian-github — APPROVED** [review]
LGTM

**Comment by jinge90** [comment]
Hi, @pvchupin 
Kind ping.

**Comment by pvchupin** [comment]
@jinge90, please address @romanovvlad comments. And please get @intel/llvm-reviewers-runtime approval.

**Comment by jinge90** [comment]
Hi, @romanovvlad 
As we discussed, we can proceed with this PR and I will submit a new PR to update the doc, is it OK?

@intel/llvm-reviewers-runtime 
Could you help review this PR?

Thanks very much.

**Comment by romanovvlad** [comment]
> Hi, @romanovvlad As we discussed, we can proceed with this PR and I will submit a new PR to update the doc, is it OK?
> 
> @intel/llvm-reviewers-runtime Could you help review this PR?
> 
> Thanks very much.

This is OK.

**Comment by jinge90** [comment]
> > Could you please add more details to the PR summary?
> 
> @jinge90, please do this.

Done.

Thanks very much.

**Comment by jinge90** [comment]
Hi, @intel/llvm-reviewers-runtime 
Could you take a look at this PR?

Thanks very much.

**Comment by jinge90** [comment]
Hi, @intel/llvm-reviewers-runtime and @sergey-semenov 
Kind ping~

**Review by cperkinsintel — APPROVED** [review]
LGTM

**Pull request merged**